### ISSUE TITLE
Unable to load any of {libcudnn_cnn.so.9.1.0, libcudnn_cnn.so.9.1, libcudnn_cnn.so.9, libcudnn_cnn.so}

### ISSUE THREAD

**Issue by Leandrocnf** [m-bain/whisperX#902]
Please, help me. I don't know much about programming, but I use whisperx via google colab because it helps me a lot with my work subtitling, but yesterday it started giving me this message:

/usr/local/lib/python3.10/dist-packages/pyannote/audio/utils/reproducibility.py:74: ReproducibilityWarning: TensorFloat-32 (TF32) has been disabled as it might lead to reproducibility issues and lower accuracy.
It can be re-enabled by calling
   >>> import torch
   >>> torch.backends.cuda.matmul.allow_tf32 = True
   >>> torch.backends.cudnn.allow_tf32 = True
See https://github.com/pyannote/pyannote-audio/issues/1370 for more details.

  warnings.warn(
Unable to load any of {libcudnn_cnn.so.9.1.0, libcudnn_cnn.so.9.1, libcudnn_cnn.so.9, libcudnn_cnn.so}
Invalid handle. Cannot load symbol cudnnCreateConvolutionDescriptor

I don't know what happened. I didn't change anything. I always use a simple command, it's just:

! whisperx (path to the videofile) --model large-v3 --output_dir (path where the file will be saved) --language en --output_format srt

I tried using another model, but it's still not working and I can't really understand the error message to try to fix whatever the problem is. If anyone could help me, I really need this to make my work easier.

**Comment by chriskathumbi** [comment]
Same issue here. Started today

**Comment by DSobrino** [comment]
I encountered the same issue. A temporary fix can be found here: https://github.com/SYSTRAN/faster-whisper/issues/1083

In my case, I added the following to my requirements.txt, and it is working now:

`ctranslate2>=4.0,<=4.4.0`

**Comment by Leandrocnf** [comment]
I solved by downgrading ctranslate 2. I used this command:

pip install ctranslate2==4.4.0

**Comment by chriskathumbi** [comment]
This works. Thanks

**Comment by MatinEbrahimkhani** [comment]
i tried downgrading as @Leandrocnf said but I still get this error massage can you please help me I am really desprate.

```
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/core/io.py:43: UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  torchaudio.set_audio_backend("soundfile")
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/pipelines/speaker_verification.py:43: UserWarning: torchaudio._backend.get_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  backend = torchaudio.get_audio_backend()
INFO:speechbrain.utils.quirks:Applied quirks (see `speechbrain.utils.quirks`): [allow_tf32, disable_jit_profiling]
INFO:speechbrain.utils.quirks:Excluded quirks specified by the `SB_DISABLE_QUIRKS` environment (comma-separated list): []
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/pipelines/speaker_verification.py:45: UserWarning: Module 'speechbrain.pretrained' was deprecated, redirecting to 'speechbrain.inference'. Please update your script. This is a change from SpeechBrain 1.0. See: https://github.com/speechbrain/speechbrain/releases/tag/v1.0.0
  from speechbrain.pretrained import (
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/pipelines/speaker_verification.py:53: UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  torchaudio.set_audio_backend(backend)
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/tasks/segmentation/mixins.py:37: UserWarning: `torchaudio.backend.common.AudioMetaData` has been moved to `torchaudio.AudioMetaData`. Please update the import path.
  from torchaudio.backend.common import AudioMetaData
Dubbyar is starting . . .
INFO:A_extract_audio.ffmpeg_extractor:Extracting audio from ./Learn-English.mp4 using FFmpeg
INFO:A_extract_audio.ffmpeg_extractor:Audio extraction successful: ./temp/Learn-English_1730473406/audio.wav
INFO:C_transcribe_audio.whisperxcom_transcriber:Transcribing ./temp/Learn-English_1730473406/audio.wav using WhisperX...
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/core/io.py:43: UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  torchaudio.set_audio_backend("soundfile")
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/pipelines/speaker_verification.py:43: UserWarning: torchaudio._backend.get_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  backend = torchaudio.get_audio_backend()
INFO:speechbrain.utils.quirks:Applied quirks (see `speechbrain.utils.quirks`): [disable_jit_profiling, allow_tf32]
INFO:speechbrain.utils.quirks:Excluded quirks specified by the `SB_DISABLE_QUIRKS` environment (comma-separated list): []
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/pipelines/speaker_verification.py:45: UserWarning: Module 'speechbrain.pretrained' was deprecated, redirecting to 'speechbrain.inference'. Please update your script. This is a change from SpeechBrain 1.0. See: https://github.com/speechbrain/speechbrain/releases/tag/v1.0.0
  from speechbrain.pretrained import (
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/pipelines/speaker_verification.py:53: UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  torchaudio.set_audio_backend(backend)
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/tasks/segmentation/mixins.py:37: UserWarning: `torchaudio.backend.common.AudioMetaData` has been moved to `torchaudio.AudioMetaData`. Please update the import path.
  from torchaudio.backend.common import AudioMetaData
Lightning automatically upgraded your loaded checkpoint from v1.5.4 to v2.4.0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint ../../../.cache/torch/whisperx-vad-segmentation.bin`
Model was trained with pyannote.audio 0.0.1, yours is 3.1.1. Bad things might happen unless you revert pyannote.audio to 0.x.
Model was trained with torch 1.10.0+cu102, yours is 2.5.1+cu124. Bad things might happen unless you revert torch to 1.x.
>>Performing transcription...
/home/iasbs/.pyenv/versions/dubenv1/lib/python3.10/site-packages/pyannote/audio/utils/reproducibility.py:74: ReproducibilityWarning: TensorFloat-32 (TF32) has been disabled as it might lead to reproducibility issues and lower accuracy.
It can be re-enabled by calling
   >>> import torch
   >>> torch.backends.cuda.matmul.allow_tf32 = True
   >>> torch.backends.cudnn.allow_tf32 = True
See https://github.com/pyannote/pyannote-audio/issues/1370 for more details.

  warnings.warn(
Could not load library libcudnn_ops_infer.so.8. Error: libcudnn_ops_infer.so.8: cannot open shared object file: No such file or directory
Aborted (core dumped)
```

**Comment by Jenniuss** [comment]
Thanks, @Leandrocnf, it works!!!

**Comment by ak4off** [comment]
didn't work

**Comment by jon-fox** [comment]
Didn't work for me either :/

ctranslate2==4.4.0
whisperx==3.1.5

**Comment by bhautik-pithadiya** [comment]
This error occurred because your cuDNN file is not set in the environment.

1. Ensure cuDNN is Installed Correctly
 Verify that the cuDNN library files are present on your system. They are usually located in the directory where CUDA is installed. For example:
 -> /usr/lib/x86_64-linux-gnu/ (Debian-based systems)
 -> /usr/local/cuda/lib64/

    Run the following command to locate cuDNN files:
```
    find /usr -name 'libcudnn*'
```
If the files are missing, you need to reinstall cuDNN.

2. Check Library Paths
Ensure the paths to the cuDNN libraries are correctly set in your environment variables:
```
export LD_LIBRARY_PATH=/usr/local/cuda/lib64:$LD_LIBRARY_PATH
export LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu:$LD_LIBRARY_PATH
```
Copy these lines and add them to ~/.bashrc or ~/.zshrc. Then run:

    source ~/.bashrc

These steps worked for me.

**Comment by jon-fox** [comment]
Yeah I've tried these with no luck. My app was working for a few months with whisperx until the lib updates.

I ended up just switching over to using faster whisper and am not having any issues.

**Comment by GUUser91** [comment]
I was able to fix this problem while using ctranslate2 version 4.5.0. First I create a virtual python environment.
```
python3.11 -m venv venv
```
Then I activate the environment.
```
source /home/user/WhisperX/venv/bin/activate
```
I install whisperx
```
pip install git+https://github.com/m-bain/whisperx.git
```
Install ctranslate2 4.5.0
```
pip install ctranslate2==4.5.0
```
Then I export the cuDNN libraries that are located in the virtual python enviroment
```
export LD_LIBRARY_PATH=/home/user/WhisperX/venv/lib/python3.11/site-packages/nvidia/cudnn/lib/
```
Now I can launch it.
```
whisperx --model large-v3 --language en The.Prime.of.Miss.Jean.Brodie.1969.flac
```
I open the bashrc file to create a alias command so I make a shorcut out of it.
```
wx='source /home/user/WhisperX/venv/bin/activate && export LD_LIBRARY_PATH=/home/user/WhisperX/venv/lib/python3.11/site-packages/nvidia/cudnn/lib/ && whisperx --model large-v3 --language en'
```
I'm using torch and torchaudio version 2.6.0 and cuda 12.8.0-1 on my artix linux distro.
